### PR TITLE
Rename extension charts to linkerd-*

### PR DESCRIPTION
**Pull request by adleong**
For consistency we rename the extension charts to a common naming scheme:

linkerd-viz -> linkerd-viz (unchanged)
jaeger -> linkerd-jaeger
linkerd2-multicluster -> linkerd-multicluster
linkerd2-multicluster-link -> linkerd-multicluster-link

We also make the chart files and chart readmes a bit more uniform.

Signed-off-by: Alex Leong <alex@buoyant.io>

### CODE REVIEW

**Review by kleimkuhler — COMMENTED** [review]
Looks good! The command examples in multicluster `install.go` and `link.go` need to be updated from `linkerd2-multicluster/README.md` to `linkerd-multicluster/README.md`

**Review by dadjeibaah — COMMENTED** [review]
Ditto to @kleimkuhler's comment. There are a couple of url links that need to be updated to match the rename to `linkerd-multicluster`. I also left one comment as well.

File: .github/workflows/integration_tests.yml [hunk 1]
```diff
@@ -40,8 +40,8 @@ jobs:
         restore-keys: |
           ${{ runner.os }}-buildx-${{ matrix.target }}-
     - name: Build docker images
-      env:
-        DOCKER_TRACE: 1
+      #env:
+      #  DOCKER_TRACE: 1
```
[the remark this answers is not in the document]
Comment (alpeb) [reply .github/workflows/integration_tests.yml]: We had this in place yesterday as a workaround for the GH actions not being able to log output to /std/error anymore, for some reason. Anyways, #5583 has already addressed the problem, but DOCKER_TRACE is no longer used, so these lines should be removed.

**Review by kleimkuhler — APPROVED** [review]

**Review by dadjeibaah — APPROVED** [review]
LGTM! 📦 🚢

**Pull request merged**